src/lib/utils/testing/custom-matchers.ts

```diff
@@ -148,76 +155,108 @@ export const customMatchers: jasmine.CustomMatcherFactories = {
     };
   },
 
-  toHaveStyle: function () {
+  toHaveAttributes: function () {
     return {
-      compare: function (actual: any, styles: { [k: string]: string } | string) {
-        let found = { }, computed = getComputedStyle(actual);
-        let allPassed: boolean = Object.keys(styles).length !== 0;
-        Object.keys(styles).forEach(prop => {
-          allPassed = allPassed && _.hasStyle(actual, prop, styles[prop], false);
-          if ( !allPassed ) {
-            found[prop] = computed.getPropertyValue(prop);
-          }
+      compare: function (actual: any, map: { [k: string]: string }) {
+        let allPassed: boolean;
+        let attributeNames = Object.keys(map);
+        allPassed = attributeNames.length !== 0;
+        attributeNames.forEach(name => {
+          allPassed = allPassed && _.hasAttribute(actual, name)
+              && _.getAttribute(actual, name) === map[name];
         });
-
         return {
           pass: allPassed,
           get message() {
-            const expectedValueStr = typeof styles === 'string' ? styles : JSON.stringify(styles);
             return `
-              Expected ${JSON.stringify(found)} ${!allPassed ? ' ' : 'not '} to contain the
-              CSS ${typeof styles === 'string' ? 'property' : 'styles'} '${expectedValueStr}'
+              Expected ${actual.outerHTML} ${allPassed ? 'not ' : ''} attributes to contain
+              '${JSON.stringify(map)}'
             `;
           }
         };
       }
     };
   },
 
-  toHaveCssStyle: function () {
+  /**
+   * Check element's inline styles only
+   */
+  toHaveStyle: function () {
     return {
-      compare: function (actual: any, styles: { [k: string]: string } | string) {
-        let allPassed: boolean;
-        if (typeof styles === 'string') {
-          allPassed = _.hasStyle(actual, styles);
-        } else {
-          allPassed = Object.keys(styles).length !== 0;
-          Object.keys(styles).forEach(prop => {
-            allPassed = allPassed && hasPrefixedStyles(actual, prop, styles[prop]);
-          });
-        }
+      compare: buildCompareStyleFunction(true)
+    };
+  },
 
-        return {
-          pass: allPassed,
-          get message() {
-            const expectedValueStr = typeof styles === 'string' ? styles : JSON.stringify(styles);
-            return `
-              Expected ${actual.outerHTML} ${!allPassed ? ' ' : 'not '} to contain the
-              CSS ${typeof styles === 'string' ? 'property' : 'styles'} '${expectedValueStr}'
-            `;
-          }
-        };
-      }
+
+  /**
+   * Check element's css stylesheet only (if not present inline)
+   */
+  toHaveCSS: function () {
+    return {
+      compare: buildCompareStyleFunction(false)
     };
   }
+
 };
 
 /**
+ * Curried value to function to check styles that are inline or in a stylesheet for the
+ * specified DOM element.
+ */
+function buildCompareStyleFunction(inlineOnly = true) {
```

Comment:
It seems like these changes are repeated between this PR, #394 and #396. Is this intentional?

Reply:
This ^ is because PR 391 will be rebased after #390 gets merged.